tools/single_use/SaveUserIDToInstrumentData.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+  * RUN-TIME WARNING FOR PROJECTS: It can take up to a whole day to run this script. It took 14 hours to run the script on the CCNA DB during testing.
+  *
+  * This script is intended for a one-time use only to restore the value of the
+ * `UserID` column of instrument tables and the `UserID` key of the instrument
+ * JSON `Data` in the flag table.
+ *
+ * This tool queries data from the `history` table and meaningfully imports
+ * data from its `userID` column back into the instrument and `flag` tables.
+ *
+ * Example use:
+ * $ php SaveUserIDToInstrumentData.php [confirm]
+
+ * PHP Version 7
+ *
+ * @category Tools
+ * @package  Loris
+ * @author   Zaliqa Rosli <zaliqa.rosli@mcin.ca>
+ * @licence  LORIS License
+ * @link     https://www.github.com/aces/Loris
+ */
+require_once __DIR__ . '/../generic_includes.php';
+
+$confirm = false;
+if (isset($argv[1]) && $argv[1] === 'confirm') {
+    $confirm = true;
+}
+
+echo "\n\nQuerying data...\n\n";
+
+// Get history DB
+$DBInfo = $config->getSetting('database');
+$hDB = isset($DBInfo['historydb']) ? $DBInfo['historydb'] : $DBInfo['database'];
+
+$result_count = 1;
```

Comment:
We're not allowed to have 0 problems?

I think this can be moved to before the foreach loop on ~line 94 and initialized to 0 if the increment is moved to the start of the loop body instead of the end..

Reply:
result_count is only used for display on the front end, never to check how many problems there are... it doesnt make much of a difference

Reply:
no, it doesn't make much difference, it's just confusing when it's initialized at the top of the script to "1".

Reply:
the result count doesn't really need to be there, but it helps to verify the numbers you are expecting when testing or running the script for real